### PR TITLE
U2F over NFC

### PR DESCRIPTION
**Pull request by RvanBaarle**
# What's new

- The ability to use U2F via NFC, instead of just over USB
- A load of magic numbers, and some other issues (which I'll fix before I remove the draft status)

# Verification 

- [ Describe how to verify changes ]

# Checklist (For Reviewer)

- [ ] PR has description of feature/bug or link to Confluence/Jira task
- [ ] Description contains actions to verify feature/bugfix
- [ ] I've built this code, uploaded it to the device and verified feature/bugfix

### CODE REVIEW

File: applications/u2f/u2f_nfc.c [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+    uint8_t* buff_tx,
+    uint16_t* buff_tx_len,
+    uint32_t* data_type,
+    void* context) {
+    U2fNfc* u2f_nfc = context;
+    *data_type = FURI_HAL_NFC_TXRX_DEFAULT;
+    buff_tx[0] = 0x02;
+    *buff_tx_len = 0;
+    FURI_LOG_D(TAG, "Something detected of length %d", buff_rx_len);
+    if(buff_rx_len >= 24) {
```
Comment (miili) [inline applications/u2f/u2f_nfc.c:43]: Reverse logic here. Test for `buff_rx_len < 24`, eliminate a logical block.

File: applications/u2f/u2f_nfc.c [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+        }
+        if(cla != 0x00) {
+            buff_tx[1] = 0x68;
+            buff_tx[2] = 0x00;
+            *buff_tx_len = 24;
+            FURI_LOG_D(TAG, "cla 0x%02x not supported", cla);
+            return true;
+        }
+        if(u2f_nfc->state == StateUnselected && ins == 0xa4) {
+            if(memcmp(&buff_rx[3], "\x04\x00\x08\xa0\x00\x00\x06\x47\x2f\x00\x01", 11) != 0) {
```
Comment (miili) [inline applications/u2f/u2f_nfc.c:63]: Can you please add more insights what the byte sequences mean? Datasheets, documentation or comments.

**Review by miili — CHANGES_REQUESTED** [review]
Awesome, learned something reading this implementation draft. Thanks for your work!

**Comment by gornekich** [comment]
Hello everyone!
Recently we released new NFC stack in https://github.com/flipperdevices/flipperzero-firmware/pull/3050 . Unfortunately, previous NFC API was not good enough to add new features like NFC over U2F. Now we finished working on new API and it is possible to make NFC over U2F application.

Thanks @RvanBaarle  for PR! I will close it for now, since we are not planning to support NFC over U2F in our firmware. However, if you want to continue working on it, you can write application and submit it to https://github.com/flipperdevices/flipper-application-catalog repository. If you have any questions, feel free to contact me on Discord.

**Comment by ChristianTacke** [comment]
Well, if the core "U2F over USB" application is part of the main flipper OS (like it currently is), then really that same application should also handle the U2F over NFC.

* It should use the same keys for both protocols. Or can an external U2F-over-NFC app use the same keys again in a proper in supported way?
* Sharing the U2F code would be really nice too. Same question: Could an external U2F-over-NFC app reuse code?

**Pull request closed without merging**